cmd/lotus-miner/sectors.go

```diff
@@ -389,112 +413,130 @@ var sectorsListCmd = &cli.Command{
 				continue
 			}
 
-			if showRemoved || st.State != api.SectorState(sealing.Removed) {
-				_, inSSet := commitedIDs[s]
-				_, inASet := activeIDs[s]
+			if !showRemoved && st.State == api.SectorState(sealing.Removed) {
+				continue
+			}
 
-				dw, vp := .0, .0
-				if st.Expiration-st.Activation > 0 {
-					rdw := big.Add(st.DealWeight, st.VerifiedDealWeight)
-					dw = float64(big.Div(rdw, big.NewInt(int64(st.Expiration-st.Activation))).Uint64())
-					vp = float64(big.Div(big.Mul(st.VerifiedDealWeight, big.NewInt(9)), big.NewInt(int64(st.Expiration-st.Activation))).Uint64())
-				}
+			_, inSSet := commitedIDs[s]
+			_, inASet := activeIDs[s]
 
-				var deals int
-				for _, deal := range st.Deals {
-					if deal != 0 {
-						deals++
+			dw, vp := .0, .0
+			estimate := st.Expiration-st.Activation <= 0
+			if !estimate {
+				rdw := big.Add(st.DealWeight, st.VerifiedDealWeight)
+				dw = float64(big.Div(rdw, big.NewInt(int64(st.Expiration-st.Activation))).Uint64())
+				vp = float64(big.Div(big.Mul(st.VerifiedDealWeight, big.NewInt(9)), big.NewInt(int64(st.Expiration-st.Activation))).Uint64())
+			} else {
+				for _, piece := range st.Pieces {
+					if piece.DealInfo != nil {
+						dw += float64(piece.Piece.Size)
+						if piece.DealInfo.DealProposal != nil && piece.DealInfo.DealProposal.VerifiedDeal {
+							vp += float64(piece.Piece.Size) * 9
```

Comment:
Can we avoid hard-coding 9 here?

Reply:
Somehow that value is really hard to get from actors, though I guess we can at least put this in a constant describing what this is